driver_test.go

```diff
@@ -0,0 +1,80 @@
+package sqlair_test
+
+import (
+	"context"
+	"database/sql"
+	"database/sql/driver"
+	"sync"
+	"unsafe"
+
+	"github.com/mattn/go-sqlite3"
+)
+
+// This file contains a wrapper sql.Driver over the SQLite driver which does
+// the real work behind the scenes. On top of it, it monitors the creation and
+// closing of prepared statements and stores the references to said statements.
+// We can later use that information to check for leaks.
+
+// The stmt registry keeps the pointers for the open and closed statements to
+// detect resource leaks. It uses pointers instead of references to the object
+// because if we stored a reference the runtime.Finalizer would not be able to
+// run.
+var openStmts map[uintptr]string = map[uintptr]string{}
+var closedStmts map[uintptr]bool = map[uintptr]bool{}
```

Comment:
I agree with you, because go-check does not say explicitly that it runs the tests sequentially it is better if we change the driver to support it.

Just like we discussed offline, I have added the test name in the DSN when creating a DB connection and that gets propagated to the statement opening and closing functions.